lib/LMSManagers/LMSNetworkManager.php

```diff
@@ -879,4 +881,71 @@ public function getFirstFreeAddress($netid)
         }
         return $ip;
     }
+
+    public function GetVlanList()
+    {
+        return $this->db->GetAll(
+            'SELECT vl.id, vlanid, description, customerid, ' . $this->db->Concat('cv.lastname', "' '", 'cv.name') . ' AS customername
+            FROM vlans AS vl
+            LEFT JOIN customers cv ON (vl.customerid = cv.id)
+            ORDER BY vlanid'
+        );
+    }
+
+    public function GetVlanInfo($id)
+    {
+        return $this->db->GetRow(
+            'SELECT vl.id, vlanid, description, customerid, ' . $this->db->Concat('cv.lastname', "' '", 'cv.name') . ' AS customername
+            FROM vlans AS vl
+            LEFT JOIN customers cv ON (vl.customerid = cv.id)
+            WHERE vl.id = ?',
+            array($id)
+        );
+    }
+
+    public function AddVlan($args)
+    {
+        $args = array(
+            'vlanid' => $args['vlanid'],
+            'description' => $args['description'],
+            'customerid' => !empty($args['customerid']) ? $args['customerid'] : null,
+        );
+
+        $result = $this->db->Execute('INSERT INTO vlans (vlanid, description, customerid) VALUES (?, ?, ?)', array($args['vlanid'], $args['description'], $args['customerid']));
+
+        if ($result && $this->syslog) {
+            $this->syslog->AddMessage(SYSLOG::RES_VLAN, SYSLOG::OPER_ADD, $args);
+        }
+        return $result;
+    }
+
+    public function DeleteVlan($id)
+    {
+        $result = $this->db->Execute('DELETE FROM vlans WHERE id = ?', array($id));
+
+        if ($result && $this->syslog) {
+            $this->syslog->AddMessage(SYSLOG::RES_VLAN, SYSLOG::OPER_DELETE, $args);
+        }
+
+        return $result;
+    }
+
+    public function UpdateVlan($args)
+    {
+        if (!empty($args['id'])) {
+            $args = array(
+                'id' => $args['id'],
+                'vlanid' => isset($args['vlanid']) ? $args['vlanid'] : null,
```

Comment:
To będzie logować w `syslog` bez kluczy, a kluczem na pewno jest id vlanu...

Reply:
Rozbudowałem komunikaty w syslogu